libcontainer/nsenter/cloned_binary.c

```diff
@@ -373,8 +432,18 @@ static int make_execfd(int *fdtype)
 static int seal_execfd(int *fd, int fdtype)
 {
 	switch (fdtype) {
-	case EFD_MEMFD:
-		return fcntl(*fd, F_ADD_SEALS, RUNC_MEMFD_SEALS);
+	case EFD_MEMFD:{
+			/*
+			 * Try to seal with newer seals, but we ignore errors because older
+			 * kernels don't support some of them. For container security only
+			 * RUNC_MEMFD_MIN_SEALS are strictly required, but the rest are
```

Comment:
Seals seem added in Linux 3.17, but we still need to support kernel 3.10 at least until the EOL of CentOS 7.

Reply:
(Slightly off-topic, but we have to have an official documentation to clarify the minimum supported kernel version)

Reply:
The first version of `memfd_create()` was added in 3.17, the seals were part of the initial implementation. We handle older kernel versions with `O_TMPFILE` (3.11 and later) or `mkostemp(3)` (glibc) for ancient kernels.

This PR doesn't change any of this behaviour, it just removes bindfd and passes `MFD_EXEC` as well as applying the two newer seals but ignores errors when adding them because they're not necessary.